src/main/java/com/box/sdk/BoxFolder.java

```diff
@@ -493,6 +493,10 @@ public void canUpload(String name, long fileSize) {
             fieldJSON.add("content_modified_at", BoxDateFormat.format(uploadParams.getModified()));
         }
 
+        if (uploadParams.getSHA1() != null) {
```

Comment:
Better to check that string is not empty
uploadParams.getSHA1() != null && !uploadParams.getSHA1().isEmpty()